src/lib/services/feature-toggle-service.ts

```diff
@@ -1287,22 +1312,51 @@ class FeatureToggleService {
                 featureName,
                 environment,
             );
-            const projectEnvironmentDefaultStrategy =
-                await this.projectStore.getDefaultStrategy(
-                    project,
-                    environment,
-                );
+            const hasDisabledStrategies = strategies.some(
+                (strategy) => strategy.disabled,
+            );
 
-            const strategy =
+            if (
```

Comment:
This code needs very detailed tests. It increases cyclomatic complexity of our no.1 hotspot